deepspeed/runtime/zero/stage3.py

```diff
@@ -840,6 +843,7 @@ def _create_fp32_partitions(self):
                     else:
                         unpinned_fp32_buffer = torch.empty(num_elements, device=self.device, dtype=torch.float)
                         self._swap_in_sub_group_to_flat_buffer(unpinned_fp32_buffer, i)
+                        self.fp32_partitioned_groups_flat[i].ds_id = ds_id
```

Comment:
There is currently redundancy between L846 and L872. To fix this, how about move L846 to after L831 for the `if ...` and move L846 into the `else ...`